File: src/main/java/gent/zeus/guitar/AppController.java
package gent.zeus.guitar;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.servlet.view.RedirectView;

import java.util.Base64;

@RestController
public class AppController {

    @GetMapping("/")
    public RedirectView rootRedirect() {
        return new RedirectView("/index");
    }

    @GetMapping("/index")
    public HelloResponse index() {
        return new HelloResponse("weeee!!");
    }

    @GetMapping("/echo")
    public HelloResponse echo(@RequestParam String message, @RequestParam(defaultValue = "1") int times) {
        return new HelloResponse(message.repeat(times));
    }

    @GetMapping("/saturn")
    public HelloResponse saturn(@RequestParam String password) {
        String encPassword = new String(Base64.getEncoder().encode(password.getBytes()));
        System.out.println(encPassword);
        if (encPassword.equals("Y29yZG9uLWJsdWU=")) {
            return new HelloResponse("Welcome to the cool zone 8)");
        } else {
            return new HelloResponse("You are not cool enough :(");
        }
    }

    public record HelloResponse(
            String response
    ) {
    }
}

File: src/main/java/gent/zeus/guitar/AppController.kt
package gent.zeus.guitar

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.servlet.view.RedirectView
import java.util.*

@RestController
class AppController {
    @GetMapping("/")
    fun rootRedirect(): RedirectView {
        return RedirectView("/index")
    }

    @GetMapping("/index")
    fun index(): HelloResponse {
        return HelloResponse("weeee!!")
    }

    @GetMapping("/echo")
    fun echo(@RequestParam message: String, @RequestParam(defaultValue = "1") times: Int): HelloResponse {
        return HelloResponse(message.repeat(times))
    }

    @GetMapping("/saturn")
    fun saturn(@RequestParam password: String): HelloResponse {
        val encPassword = String(Base64.getEncoder().encode(password.toByteArray()))
        println(encPassword)
        if (encPassword == "Y29yZG9uLWJsdWU=") {
            return HelloResponse("Welcome to the cool zone 8)")
        } else {
            return HelloResponse("You are not cool enough :(")
        }
    }

    @JvmRecord
    data class HelloResponse(
        val response: String?
    )
}
